src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -116,27 +115,27 @@ Status CoreWorkerDirectActorTaskSubmitter::SubscribeActorUpdates() {
 
 void CoreWorkerDirectActorTaskSubmitter::ConnectAndSendPendingTasks(
     const ActorID &actor_id, std::string ip_address, int port) {
-  std::unique_ptr<rpc::DirectActorClient> grpc_client(
-      new rpc::DirectActorClient(ip_address, port, client_call_manager_));
+  std::shared_ptr<rpc::DirectActorClient> grpc_client =
+      rpc::DirectActorClient::make(ip_address, port, client_call_manager_);
   RAY_CHECK(rpc_clients_.emplace(actor_id, std::move(grpc_client)).second);
 
   // Submit all pending requests.
   auto &client = rpc_clients_[actor_id];
   auto &requests = pending_requests_[actor_id];
   while (!requests.empty()) {
-    const auto &request = *requests.front();
-    PushTask(*client, request, TaskID::FromBinary(request.task_spec().task_id()),
-             request.task_spec().num_returns());
+    auto request = std::move(requests.front());
+    PushTask(*client, std::move(request),
```

Comment:
Did you mean to call `std::move` twice?

Reply:
I think this is required. It looks suspicious to move an argument that we also use in other args though, so cleaned this up a bit.